api_tests/crossref/views/test_crossref_email_response.py

```diff
@@ -0,0 +1,160 @@
+                msg='{}{}'.format(
+                    response_data['timestamp'],
+                    response_data['token']
+                ),
+                digestmod=hashlib.sha256,
+            ).hexdigest(),
+        }
+        data.update(response_data)
+        data = {
+            key: value
+            for key, value in data.iteritems()
+            if value is not None
+        }
+        return data
+
+    @pytest.fixture()
+    def preprint(self):
+        return factories.PreprintFactory(set_doi=False)
+
+    @pytest.fixture()
+    def error_xml(self, preprint):
+        return """
+        <?xml version="1.0" encoding="UTF-8"?>
+        <doi_batch_diagnostic status="completed" sp="cs3.crossref.org">
+           <submission_id>1390675109</submission_id>
+           <batch_id>{}</batch_id>
+           <record_diagnostic status="Failure">
+              <doi />
+              <msg></msg>
+           </record_diagnostic>
+           <batch_data>
+              <record_count>1</record_count>
+              <success_count>0</success_count>
+              <warning_count>0</warning_count>
+              <failure_count>1</failure_count>
+           </batch_data>
+        </doi_batch_diagnostic>
+        """.format(preprint._id)
+
+    @pytest.fixture()
+    def success_xml(self, preprint):
+        return """
+            <?xml version="1.0" encoding="UTF-8"?>
+            <doi_batch_diagnostic status="completed" sp="cs3.crossref.org">
+               <submission_id>1390675475</submission_id>
+               <batch_id>{}</batch_id>
+               <record_diagnostic status="Success">
+                  <doi>10.31219/FK2OSF.IO/{}</doi>
+                  <msg>Successfully added</msg>
+               </record_diagnostic>
+               <batch_data>
+                  <record_count>1</record_count>
+                  <success_count>1</success_count>
+                  <warning_count>0</warning_count>
+                  <failure_count>0</failure_count>
+               </batch_data>
+            </doi_batch_diagnostic>
+        """.format(preprint._id, preprint._id)
+
+    @pytest.fixture()
+    def update_success_xml(self, preprint):
+        return """
+        <?xml version="1.0" encoding="UTF-8"?>
+        <doi_batch_diagnostic status="completed" sp="cs3.crossref.org">
+            <submission_id>1390757455</submission_id>
+            <batch_id>{}</batch_id>
+            <record_diagnostic status="Success">
+                <doi>10.31219/FK2osf.io/{}</doi>
+                <msg>Successfully updated</msg>
+            </record_diagnostic>
+            <batch_data>
+                <record_count>1</record_count>
+                <success_count>1</success_count>
+                <warning_count>0</warning_count>
+                <failure_count>0</failure_count>
+            </batch_data>
+        </doi_batch_diagnostic>
+        """.format(preprint._id, preprint._id)
+
+    def mailgun_response(self, response):
```

Comment:
This is a misnomer: this is the request payload, not the response payload, right? How about `make_mailgun_payload`?

Reply:
Ah, I see that `create_mailgun_request_context` is what makes an actual mailgun payload, so maybe rename *that* to `make_mailgun_payload` and remove this method.